integration/k3t/cluster.go

```diff
@@ -0,0 +1,132 @@
+			},
+		},
+	}
+
+	clusterConfig, err := config.TransformSimpleToClusterConfig(ctx, runtimes.SelectedRuntime, simpleCfg)
+	require.NoError(t, err, "Can't transform simple config to cluster config")
+
+	clusterConfig, err = config.ProcessClusterConfig(*clusterConfig)
+	require.NoError(t, err, "Failed to process cluster config")
+
+	err = config.ValidateClusterConfig(ctx, runtimes.SelectedRuntime, *clusterConfig)
+	require.NoError(t, err, "Failed to validate cluster config")
+
+	// Check existing cluster.
+	_, err = k3dclient.ClusterGet(ctx, runtimes.SelectedRuntime, &clusterConfig.Cluster)
+	require.Errorf(t, err, "Cluster %s already exists, remove it running `k3d cluster delete %s`, or by just removing the corresponding docker containers.", clusterConfig.Name, clusterConfig.Name)
+
+	// Prepare cleanup.
+	t.Cleanup(func() {
+		err := k3dclient.ClusterDelete(ctx, runtimes.SelectedRuntime, &clusterConfig.Cluster, k3d.ClusterDeleteOpts{SkipRegistryCheck: true})
+		require.NoError(t, err, "Can't delete cluster on cleanup.")
+	})
+
+	// Create the cluster.
+	err = k3dclient.ClusterRun(ctx, runtimes.SelectedRuntime, clusterConfig)
+	require.NoError(t, err, "Failed creating cluster.")
+	t.Logf("Cluster '%s' created successfully!", clusterConfig.Cluster.Name)
+
+	// Get kubeconfig and instantiate kubernetes.Clientset.
+	kubeConfigFile := filepath.Join(t.TempDir(), fmt.Sprintf("kubeconfig.%s.yaml", clusterName))
+
+	kubeconfig, err := k3dclient.KubeconfigGet(ctx, runtimes.SelectedRuntime, &clusterConfig.Cluster)
+	require.NoError(t, err, "Failed to get kubeconfig from cluster")
+
+	err = clientcmd.WriteToFile(*kubeconfig, kubeConfigFile)
+	require.NoError(t, err, "Failed to write kubeconfig to file")
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfigFile)
+	require.NoError(t, err, "Failed to build config from flags")
+
+	clientset, err := kubernetes.NewForConfig(config)
+	require.NoError(t, err, "Failed to build kubernetes clientset")
+
+	t.Logf("KUBECONFIG=%s", kubeConfigFile)
+
+	return Cluster{
+		k:             clientset,
+		clusterConfig: clusterConfig,
+		lbPort:        lbPort,
+	}
+}
+
+func getFreePort(t *testing.T, defaultPort string) string {
+	port, err := cliutil.GetFreePort()
+	hostAPIPort := strconv.Itoa(port)
+	if err != nil || port == 0 {
+		t.Logf("Failed to get random free port: %+v", err)
+		t.Logf("Falling back to internal port %s (may be blocked though)...", k3d.DefaultAPIPort)
+		return defaultPort
+	}
+	return hostAPIPort
+}
+
+type Cluster struct {
+	clusterConfig *v1alpha4.ClusterConfig
+	k             *kubernetes.Clientset
+	lbPort        string
+}
+
+func (c Cluster) API() *kubernetes.Clientset {
+	return c.k
+}
+
+func (c Cluster) LBPort() string {
+	return c.lbPort
+}
+
+func (c Cluster) LoadImages(ctx context.Context, t *testing.T, images ...string) {
+	t.Helper()
+	// TODO: we're using k3d.ImportModeTooldNode mode because the auto-detect (that uses pipe) fails in CI: https://github.com/k3d-io/k3d/issues/900
```

Comment:
[nit]
```suggestion
	// TODO: we're using k3d.ImportModeToolsNode mode because the auto-detect (that uses pipe) fails in CI: https://github.com/k3d-io/k3d/issues/900
```

Reply:
Done in https://github.com/grafana/rollout-operator/pull/30/commits/1cb4c73912501c7d075d2490c83c92a49fccbc7c